repository/src/main/java/org/pentaho/platform/repository2/unified/jcr/JcrRepositoryFileDao.java

```diff
@@ -657,15 +657,16 @@ public Object doInJcr( final Session session ) throws RepositoryException, IOExc
           RepositoryFileAcl toBeDeletedFileAcl = aclDao.getAcl( fileToBeDeleted.getId() );
           // Invoke accessVoterManager to see if we have access to perform this operation
           if ( !accessVoterManager.hasAccess( fileToBeDeleted, RepositoryFilePermission.DELETE, toBeDeletedFileAcl,
-              PentahoSessionHolder.getSession() ) ) {
+                  PentahoSessionHolder.getSession() ) ) {
             return null;
           }
-        }
-        List<RepositoryFilePermission> perms = new ArrayList<RepositoryFilePermission>();
-        perms.add( RepositoryFilePermission.DELETE );
-        if ( !aclDao.hasAccess( fileToBeDeleted.getPath(), EnumSet.copyOf( perms ) ) ) {
-          throw new AccessDeniedException( Messages.getInstance().getString(
-              "JcrRepositoryFileDao.ERROR_0006_ACCESS_DENIED_DELETE", fileId ) );
+
+          List<RepositoryFilePermission> perms = new ArrayList<RepositoryFilePermission>();
+          perms.add( RepositoryFilePermission.DELETE );
+          if ( !aclDao.hasAccess( fileToBeDeleted.getPath(), EnumSet.copyOf( perms ) ) ) {
```

Comment:
Shouldn't we be estingi if fileToBeDeleted is null here?

Reply:
@pamval filesToBeDeleted here is also enclosed by the if(filesToBeDeleted != null)

Reply:
@pamval, see line 656 above.